src/Plugin/Visualisation/Style/TableVisualisationStyleBase.php

```diff
@@ -63,6 +64,14 @@ protected function tableBuildSettings(array $records) {
    */
   abstract protected function rowHeaderField();
 
+  /**
+   * Gets the table options.
+   *
+   * @return array
+   *   The table options.
+   */
+  abstract protected function getTableOptions();
```

Comment:
This doesn't make any sense here because `TableVisualisationStyleBase.php` doesn't have a settings form. Could you please remove it?